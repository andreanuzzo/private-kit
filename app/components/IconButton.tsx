import React from 'react';
import { SvgXml } from 'react-native-svg';
import { Colors } from '../styles';
import { TouchableOpacity } from 'react-native';

interface IconButtonProps {
  icon: string;
  accessibilityLabel?: string;
  size?: number;
  onPress?: () => void;
  color?: string;
}

export const IconButton = ({
  icon,
  accessibilityLabel,
  size,
  color,
  ...otherProps
}: IconButtonProps): JSX.Element => {
  return (
    <TouchableOpacity
      style={{
        alignItems: 'center',
        alignContent: 'center',
      }}
      accessibilityLabel={accessibilityLabel}
      {...otherProps}>
      <SvgXml
        color={color || Colors.icon}
        xml={icon}
        width={size || 24}
        height={size || 24}
      />
    </TouchableOpacity>
  );
};
