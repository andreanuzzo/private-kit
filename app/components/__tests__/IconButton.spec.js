import { render } from '@testing-library/react-native';
import React from 'react';

import { IconButton } from '../IconButton';

const svg = `<g />`;

it('renders the icon in a touchable opacity', () => {
  const { asJSON } = render(
    <IconButton icon={svg} accessibilityLabel='Label' />,
  );

  expect(asJSON()).toMatchSnapshot();
});

it('allows size override', () => {
  const { asJSON } = render(<IconButton icon={svg} size={48} />);

  expect(asJSON()).toMatchSnapshot();
});
