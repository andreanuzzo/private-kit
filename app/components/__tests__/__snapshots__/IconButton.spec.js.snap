// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows size override 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
      }
    }
  >
    <SvgXml
      color="#606060"
      height={48}
      width={48}
      xml="<g />"
    />
  </TouchableOpacity>
</View>
`;

exports[`renders the icon in a touchable opacity 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    accessibilityLabel="Label"
    activeOpacity={0.2}
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
      }
    }
  >
    <SvgXml
      color="#606060"
      height={24}
      width={24}
      xml="<g />"
    />
  </TouchableOpacity>
</View>
`;
